src/test/java/it/eng/idsa/dataapp/util/MessageUtilTest.java

```diff
@@ -1,254 +1,81 @@
-		assertNull(payload);
-	}
+		messageUtil = new MessageUtil(issuerConnector);
 
-	@Test
-	public void testResponsePayload_StringContractAgreementMessage(){
-		String payload = messageUtil.createResponsePayload(UtilMessageService.getContractAgreementMessage(), null);
-		assertNull(payload);;
-	}
-	
-	//Response for ArtifactRequestMessage
-	
-	@Test
-	public void testResponsePayload_IDSArtifactRequestMessage(){
-		String payload = messageUtil.createResponsePayload(UtilMessageService.getArtifactRequestMessage(), "ABC");
-		assertTrue(payload.contains("John"));
-		assertTrue(payload.contains("Doe"));
-		assertTrue(payload.contains("591  Franklin Street, Pennsylvania"));
 	}
 
 	@Test
-	public void createArtifactResponseMessage() throws IOException  {
-		// provide default ArtifactRequestMessage so it does not fail on check for
-		// transfer contract and requested element
-		Message message = messageUtil.createArtifactResponseMessage(UtilMessageService.getArtifactRequestMessage());
-		assertNotNull(message);
-		assertTrue(message instanceof ArtifactResponseMessage);
-		serializer.serialize(message);
-	}
+	public void bigPayload() throws UnsupportedOperationException, IOException {
+		String header = UtilMessageService.getMessageAsString(UtilMessageService.getArtifactResponseMessage());
 
-	@Test
-	public void createContractAgreementMessage() throws IOException  {
-		Message message = messageUtil.createContractAgreementMessage(UtilMessageService.getContractRequestMessage());
-		assertNotNull(message);
-		assertTrue(message instanceof ContractAgreementMessage);
-		serializer.serialize(message);
+		HttpEntity httpEntity = messageUtil.createMultipartMessageForm(header, BigPayload.BIG_PAYLOAD,
+				ContentType.TEXT_PLAIN);
+		assertNotNull(httpEntity);
+
+		ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+		httpEntity.writeTo(outStream);
+		outStream.flush();
+
+		String bigMultipartResponse = new String(outStream.toByteArray(), StandardCharsets.UTF_8);
+		assertNotNull(bigMultipartResponse);
 	}
 
 	@Test
-	public void createResultMessage() throws IOException  {
-		Message message = messageUtil.createResultMessage(UtilMessageService.getArtifactRequestMessage());
-		assertNotNull(message);
-		assertTrue(message instanceof ResultMessage);
-		serializer.serialize(message);
+	public void createMultipartMessageFormContentTypeNullTest() throws UnsupportedOperationException, IOException {
+		String header = UtilMessageService.getMessageAsString(UtilMessageService.getArtifactResponseMessage());
+
+		HttpEntity httpEntity = messageUtil.createMultipartMessageForm(header, createResponsePayload(), null);
+		assertNotNull(httpEntity);
+
+		ByteArrayOutputStream outStream = new ByteArrayOutputStream();
+		httpEntity.writeTo(outStream);
+		outStream.flush();
+
+		String bigMultipartResponse = new String(outStream.toByteArray(), StandardCharsets.UTF_8);
```

Comment:
Is this one also bigMultipartResponse? ;)

Reply:
You're right, it's not, changed.